Add release notes for 0.13.0 and update version tag and date (#615)
pika-org/pika: CHANGELOG.md

<!--- Distributed under the Boost Software License, Version 1.0. (See accompanying -->
<!--- file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt) -->

## 0.13.0 (2023-03-08)

### New features

- Add better compile-time error messages to diagnose one-shot senders being used as multi-shot senders. ([#586](https://github.com/pika-org/pika/pull/586))

> - Optimize the MPI polling implementation. ([#593](https://github.com/pika-org/pika/pull/593))

Reviewer: This is unlikely to make it, but we can add still add it separately if it makes it before Wednesday.

### Breaking changes

- Remove the `PIKA_WITH_BACKGROUND_THREAD_COUNTERS` CMake option. These counters are no longer available. ([#588](https://github.com/pika-org/pika/pull/588))
- Update required stdexec commit. pika is now tested with `6510f5bd69cc03b24668f26eda3dd3cca7e81bb2` ([#597](https://github.com/pika-org/pika/pull/597))

> - Cleaned up modules and moved minor functionality into `detail` namespaces. ([#594](https://github.com/pika-org/pika/pull/594), [#595](https://github.com/pika-org/pika/pull/595), [#596](https://github.com/pika-org/pika/pull/596), [#599](https://github.com/pika-org/pika/pull/599), [#605](https://github.com/pika-org/pika/pull/605), [#607](https://github.com/pika-org/pika/pull/607))

Reviewer: ```suggestion
- Cleaned up modules and moved minor functionality into `detail` namespaces. ([#594](https://github.com/pika-org/pika/pull/594), [#595](https://github.com/pika-org/pika/pull/595), [#596](https://github.com/pika-org/pika/pull/596), [#599](https://github.com/pika-org/pika/pull/599), [#607](https://github.com/pika-org/pika/pull/607))
```

<!--- Distributed under the Boost Software License, Version 1.0. (See accompanying -->
<!--- file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt) -->

## 0.13.0 (2023-03-08)

### New features

> - Add `static_assert` to have better error messages with senders. ([#586](https://github.com/pika-org/pika/pull/586))

Reviewer: Maybe:
```suggestion
- Add better compile-time error messages to diagnose one-shot senders being used as multi-shot senders. ([#586](https://github.com/pika-org/pika/pull/586))
```
?

### Breaking changes

> - Rename `done_*` with `stopped_*` to conform with stdexec implementation. ([#583](https://github.com/pika-org/pika/pull/583))

Reviewer: This is purely internal and can be left out.
```suggestion
```

> - Remove `PIKA_WITH_BACKGROUND_THREAD_COUNTERS`. ([#588](https://github.com/pika-org/pika/pull/588))

Reviewer: ```suggestion
- Remove the `PIKA_WITH_BACKGROUND_THREAD_COUNTERS` CMake option. These counters are no longer available. ([#588](https://github.com/pika-org/pika/pull/588))
```

> - Update stdexec commit used. ([#597](https://github.com/pika-org/pika/pull/597))

Reviewer: ```suggestion
- Update required stdexec commit. pika is now tested with `6510f5bd69cc03b24668f26eda3dd3cca7e81bb2`. ([#597](https://github.com/pika-org/pika/pull/597))
```

> - Update moodycamel concurrent queue to the latest version. ([#608](https://github.com/pika-org/pika/pull/608))

Reviewer: Internal, can be removed.
```suggestion
```

> - Cleaned up modules and moved minor functionality into `detail` namespaces. ([#607](https://github.com/pika-org/pika/pull/607), [#607](https://github.com/pika-org/pika/pull/607))

Reviewer: 607 included twice? I also think you maybe meant to use the list that's further down on `fix warning about no return...`?

### Bugfixes

- Initialize HIP early to avoid concurrent initialization. ([#591](https://github.com/pika-org/pika/pull/591))

> - Minor fix in `cuda_sender_test`. ([#592](https://github.com/pika-org/pika/pull/592))

Reviewer: This can be left out since it's just the test that was fixed.
```suggestion
```

> - fix warning about no return in `unique_any_sender` connect overload. ([#594](https://github.com/pika-org/pika/pull/594), [#595](https://github.com/pika-org/pika/pull/595), [#596](https://github.com/pika-org/pika/pull/596), [#599](https://github.com/pika-org/pika/pull/599), [#605](https://github.com/pika-org/pika/pull/605))

Reviewer: Since this warning was introduced after the latest release I don't think we need to include it (it was introduced by #586).
```suggestion
```
